dataflow/src/backlog/mod.rs

```diff
@@ -231,6 +241,37 @@ impl WriteHandle {
     pub(crate) fn is_partial(&self) -> bool {
         self.partial
     }
+
+    /// Evict `count` randomly selected keys from state and return them along with the number of
+    /// bytes that will be freed once the underlying `evmap` applies the operation.
+    pub fn evict_random_key(&mut self, rng: &mut ThreadRng) -> u64 {
+        let mut bytes_to_be_freed = 0;
+        if self.mem_size > 0 {
+            match self.handle.empty_at_index(rng.gen()) {
+                None => (),
+                Some(vs) => {
+                    let size: u64 = vs.into_iter().map(|r| r.deep_size_of() as u64).sum();
+                    bytes_to_be_freed += size;
+                }
+            }
+            self.mem_size = self.mem_size
+                .checked_sub(bytes_to_be_freed as usize)
+                .unwrap();
+        }
+        bytes_to_be_freed
+    }
+}
+
+impl SizeOf for WriteHandle {
+    fn size_of(&self) -> u64 {
+        use std::mem::size_of;
+
+        size_of::<Self>() as u64
+    }
+
+    fn deep_size_of(&self) -> u64 {
+        self.mem_size as u64
```

Comment:
Should this not include the size of `Self`?

Reply:
Sure, can include it, but it's just a constant 16 (?) bytes.